types/adnexo/ax-dist/uplink.js

```diff
@@ -0,0 +1,186 @@
+    data.echo = payload;
+    topic = "lifecycle";
+  } else if (port === 2) {
+    data.payloadType = "RESTART";
+    topic = "lifecycle";
+  } else if (port === 3) {
+    data.payloadType = "NEW_CONFIGURATION_ACCEPTED";
+    topic = "lifecycle";
+  } else if (port === 4) {
+    let pointer = 0;
+
+    while (pointer < payload.length) {
+      const header = Bits.bitsToUnsigned(bits.substr(pointer * 4, 8));
+      pointer += 2;
+
+      switch (header) {
+        case 0x01:
+          data.appEui = payload.substr(pointer, 16);
+          pointer += 16;
+          break;
+        case 0x02:
+          data.appKey = payload.substr(pointer, 32);
+          pointer += 32;
+          break;
+        case 0x03:
+          data.devEui = payload.substr(pointer, 16);
+          pointer += 16;
+          break;
+        case 0x04:
+          data.measurementInterval = toLittleEndian(
+            payload.substr(pointer, 8),
+            false,
+          );
+          pointer += 8;
+          break;
+        case 0x05:
+          data.sendInterval = toLittleEndian(payload.substr(pointer, 2), false);
+          pointer += 2;
+          break;
+        case 0x06:
+          data.joinInterval = toLittleEndian(payload.substr(pointer, 8), false);
+          pointer += 8;
+          break;
+        case 0x07:
+          data.reed = toLittleEndian(payload.substr(pointer, 2), false);
+          pointer += 2;
+          break;
+        case 0x52:
+          data.thresholdDistance = toLittleEndian(
+            payload.substr(pointer, 4),
+            true,
+          );
+          pointer += 4;
+          break;
+
+        default:
+          pointer = payload.length;
+          break;
+      }
+    }
+
+    data.payloadType = "REPORT_CONFIGURATION";
+    topic = "lifecycle";
+  } else if (port === 5) {
+    const errorPort = Bits.bitsToUnsigned(bits.substr(0, 8));
+    const errorCode = Bits.bitsToUnsigned(bits.substr(0, 8));
+
+    if (errorPort === 2) {
+      data.errorOn = "RESTART";
+    } else if (errorPort === 3) {
+      data.errorOn = "PARAM_CHANGE";
+    } else if (errorPort === 4) {
+      data.errorOn = "PARAM_READ";
+    }
+
+    if (errorCode === 1) {
+      data.errorCode = "PARAM_UNKNOWN";
+    } else if (errorCode === 2) {
+      data.errorCode = "PARAM_READ_ONLY";
+    } else if (errorCode === 4) {
+      data.errorCode = "PARAM_NOT_READABLE";
+    } else if (errorCode === 8) {
+      data.errorCode = "COMMAND_STRUCTURE_WRONG";
+    }
+    topic = "error";
+  } else if (port === 100 || port === 101 || port === 102) {
+    const distance = toLittleEndian(payload.substr(0, 4), false) / 10;
+    if (distance !== 6553.5) {
+      data.distance = distance;
+    } else {
+      data.distance = null;
+    }
+
+    if (event.device !== undefined && data.distance !== undefined) {
+      if (event.device.customFields !== undefined) {
+        const scaleLength = Number(
+          event.device.customFields["Height of the tank in cm"],
```

Comment:
better use custom fields that use camelCase: tankHeightCm in order to ease programmatic access

Reply:
Yeah but it's harder to understand for the people which need to fill in those values afterwards. 
<img width="508" alt="image" src="https://user-images.githubusercontent.com/73245684/180978344-bc4d328a-c66d-47a0-b9c6-cf362a191121.png">
For me either way is fine.

Reply:
we should maybe add a description to custom fields that will allow writing sentence like you did? could you propose that in Canny?